### PR TITLE
fix(gateapi): return request body for pipline-template

### PR DESCRIPTION
**Pull request by nevins-b**
Addresses #105 there are probably other places something similar is needed. I also am unable to get the response body back from the delete call to this controller but there two are a good starting point.

### CODE REVIEW

File: gateapi/api/swagger.yaml [hunk 1]
```diff
@@ -3779,6 +3779,11 @@ paths:
       responses:
         201:
           description: "Created"
+          schema:
```
Comment (jtk54) [inline gateapi/api/swagger.yaml]: Is there a symmetric gate change for the swagger and gate API change?

File: gateapi/pipeline_templates_controller_api.go [hunk 2]
```diff
@@ -493,19 +500,24 @@ func (a *PipelineTemplatesControllerApiService) UpdateUsingPOST(ctx context.Cont
...
+		return successPayload, localVarHttpResponse, reportError("Status: %v, Body: %s", localVarHttpResponse.Status, bodyBytes)
 	}
 
-	return localVarHttpResponse, err
+	if err = json.NewDecoder(localVarHttpResponse.Body).Decode(&successPayload); err != nil {
+		return successPayload, localVarHttpResponse, err
+	}
+
+
+	return successPayload, localVarHttpResponse, err
```
Comment (jtk54) [inline gateapi/pipeline_templates_controller_api.go]: Are there no changes necessary in the commands to output the reference to the Orca op?

Comment (jtk54) [reply gateapi/pipeline_templates_controller_api.go]: Ah I see, I meant from `spin`'s commands, e.g. [pipeline-templates save command](https://github.com/spinnaker/spin/blob/master/cmd/pipeline-template/save.go#L107). Are you consuming the [gateapi](https://github.com/spinnaker/spin/tree/master/gateapi) without using `spin`?

Comment (nevins-b) [reply gateapi/pipeline_templates_controller_api.go]: Yes, I'm using it as a library. The `spin` command uses the V2 API for this so should be unaffected by these changes.

**Comment by nevins-b** [comment]
The more I look at this the more I feel like this is a bug in `swagger-codegen` since there should be a response body even without these changes according to the swagger we have.

**Comment by jtk54** [comment]
Are these changes manual or were they generated using `swagger-codegen`? If you had to manually add the return bodies, then I agree this is likely an issue either with `swagger-codegen` or with our generation process.

**Comment by nevins-b** [comment]
I manually updated the `swagger.yaml` but the Go code changes where generated by the tool. I'm just surprised that we aren't getting a response body back already since it is defined for the `202 Accepted` response

**Comment by jtk54** [comment]
The manual edits to the swagger.json shouldn't be necessary and ought to be generated through https://github.com/spinnaker/gate/blob/master/swagger/generate_swagger.sh in combination with the swagger rules at https://github.com/spinnaker/gate/blob/master/gate-web/config/gate.yml#L55.

**Comment by louisjimenez** [comment]
Closing this as it has not been updated for some time. Feel free to re-open if there are updates.

**Pull request closed without merging**